### ISSUE TITLE
Best model not automatically exported after training is done

### ISSUE THREAD

**Issue by ashleycheng** [OpenNMT/OpenNMT-tf#579]
Hi, I am training sequence to sequence mapping using the latest master code. However, when the training is done, I could not find the export directory.

Here is my command:
`onmt-main --model_type Transformer --config data.yml --auto_config train --with_eval`

And my configuration file:
```
model_dir: run/

data:
  train_features_file: train_src.txt
  train_labels_file: train_tgt.txt
  eval_features_file: val_src.txt
  eval_labels_file: val_tgt.txt
  source_vocabulary: src-vocab.txt
  target_vocabulary: tgt-vocab.txt

train:
  max_step: 10000

eval:
  steps: 200
  export_on_best: f1
```

Here is the logs:
```
INFO:tensorflow:Saved checkpoint run/ckpt-0
WARNING:tensorflow:From /usr/local/lib/python3.6/dist-packages/tensorflow_core/python/ops/resource_variable_ops.py:1781: calling BaseResourceVariable.__init__ (from tensorflow.python.ops.resource_variable_ops) with constraint is deprecated and will be removed in a future
version.
Instructions for updating:
If using Keras pass *_constraint arguments to layers.
INFO:tensorflow:Step = 100 ; source words/s = 10610, target words/s = 11749 ; Learning rate = 0.000012 ; Loss = 6.254988
INFO:tensorflow:Step = 200 ; source words/s = 16884, target words/s = 18684 ; Learning rate = 0.000025 ; Loss = 5.477633
INFO:tensorflow:Running evaluation for step 200
INFO:tensorflow:Evaluation result for step 200: loss = 5.263096 ; perplexity = 193.078400
INFO:tensorflow:Step = 300 ; source words/s = 3123, target words/s = 3455 ; Learning rate = 0.000037 ; Loss = 5.041226
INFO:tensorflow:Step = 400 ; source words/s = 16956, target words/s = 18778 ; Learning rate = 0.000050 ; Loss = 4.414327
INFO:tensorflow:Running evaluation for step 400
INFO:tensorflow:Evaluation result for step 400: loss = 3.377359 ; perplexity = 29.293308
INFO:tensorflow:Step = 500 ; source words/s = 5519, target words/s = 6108 ; Learning rate = 0.000062 ; Loss = 3.446130
INFO:tensorflow:Step = 600 ; source words/s = 16926, target words/s = 18731 ; Learning rate = 0.000074 ; Loss = 2.058619
INFO:tensorflow:Running evaluation for step 600
INFO:tensorflow:Evaluation result for step 600: loss = 0.892893 ; perplexity = 2.442183
INFO:tensorflow:Step = 700 ; source words/s = 3043, target words/s = 3370 ; Learning rate = 0.000087 ; Loss = 1.745045
INFO:tensorflow:Step = 800 ; source words/s = 16943, target words/s = 18748 ; Learning rate = 0.000099 ; Loss = 1.519234
INFO:tensorflow:Running evaluation for step 800
INFO:tensorflow:Evaluation result for step 800: loss = 0.179646 ; perplexity = 1.196793
INFO:tensorflow:Step = 900 ; source words/s = 10794, target words/s = 11943 ; Learning rate = 0.000111 ; Loss = 1.292324
```
The evaluation run every 200 steps, but It doesn't save model automatically if it achieves the best evaluation so far.

**Comment by guillaumekln** [comment]
Hi,

You configured to export on the best `f1` metric:

```yaml
eval:
  export_on_best: f1
```

However, the evaluation of your model does not produce the `f1` metric, only `loss` and `perplexity`:

```
INFO:tensorflow:Evaluation result for step 800: loss = 0.179646 ; perplexity = 1.196793
```

So you could replace `f1` by `loss` to enable the automatic export.

We should raise an error with this configuration though. I'm keeping this issue open until this is improved. Thanks.

**Comment by ashleycheng** [comment]
I changed to loss and successfully exported the best models. Really grateful for your fast response and help.
There seems not so many details about exporting metrics, maybe a little more information in the document about this part would be even better?

**Comment by guillaumekln** [comment]
There is one section in the documentation related to this: http://opennmt.net/OpenNMT-tf/serving.html#automatic-export

Surely it could (and should) be completed with more details.
